primitives/src/block.rs

```diff
@@ -365,11 +365,89 @@ mod tests {
     fn version_is_signalling() {
         let version = Version::from_consensus(0b00100000000000000000000000000010);
         assert!(Version::is_signalling_soft_fork(&version, 1));
+        let version = Version::from_consensus(0b00110000000000000000000000000000);
+        assert!(Version::is_signalling_soft_fork(&version, 28));
     }
 
     #[test]
     fn version_is_not_signalling() {
         let version = Version::from_consensus(0b00100000000000000000000000000010);
         assert!(!Version::is_signalling_soft_fork(&version, 0));
     }
+
+    #[test]
+    fn version_to_consensus() {
+        let version = Version::from_consensus(1234567890);
+        assert_eq!(version.to_consensus(), 1234567890);
+    }
+
+    // Test the creation of a dummy block
+    #[cfg(feature = "alloc")]
+    #[test]
+    fn block_creation() {
+        use crate::locktime::absolute;
+        use crate::merkle_tree::TxMerkleNode;
+        use crate::{transaction, Amount, Sequence, Witness};
+        use crate::script::ScriptBuf;
+
+        let create_dummy_tx = |txid, version, lock_time, sequence| Transaction {
+            version,
+            lock_time: absolute::LockTime::from_consensus(lock_time),
+            input: vec![transaction::TxIn {
+                previous_output: transaction::OutPoint { txid, vout: 0 },
+                script_sig: ScriptBuf::new(),
+                sequence: Sequence(sequence),
+                witness: Witness::new(),
+            }],
+            output: vec![transaction::TxOut {
+                value: Amount::ONE_SAT,
+                script_pubkey: ScriptBuf::new(),
+            }],
+        };
+
+        let txid1 =
+            transaction::Txid::from_byte_array(hashes::sha256::Hash::hash(&[1]).to_byte_array());
+        let txid2 =
+            transaction::Txid::from_byte_array(hashes::sha256::Hash::hash(&[2]).to_byte_array());
+        let tx1 = create_dummy_tx(txid1, transaction::Version::ONE, 2, 1);
+        let tx2 = create_dummy_tx(txid2, transaction::Version::TWO, 3, 2);
```

Comment:
If you are going to use magic numbers (eg 1, 2, 3) its nice if they are either obviously meaningful or obviously not meaningful. I usually throw a comment on them if I can't come up with a value that is one or the other. Here specifically an `absolute::LockTime` of block 2 is obviously not-meaningful but I had to look a bit to work that out.

Also if you can write tests so that a reader later can easily see what we are testing and why then we have more chance of finding holes in the tests.